### PR TITLE
Github action for CI job

### PR DESCRIPTION
**Pull request by davidtvs**
@NaleRaphael had to do some minor changes to the unit tests after stumbling on the page from [pytest about good integration practices](https://docs.pytest.org/en/latest/goodpractices.html) and [this StackOverflow post](https://stackoverflow.com/a/50796779).

The change is basically just removing the `__init__.py` from the tests folder, which then led me to use setup.py to install the package in editable mode to have the command `pytest` working, which then complained about relative path imports.

### CODE REVIEW

**Comment by NaleRaphael** [comment]
Thanks for pointing this out, I didn't notice this case. 🤦

While I was developing those test cases, I chose to add a `__init__.py` in `tests` directory in order to make *explicit relative imports* work and keep those import statements unambiguous. But it turns out to be an issue like the description in the pytest documentation you given:

> But now this introduces a subtle problem: in order to load the test modules from the `tests` directory, pytest prepends the root of the repository to `sys.path`, which adds the side-effect that now `mypkg` is also importable.  This is problematic if you are using a tool like `tox` to test your package in a virtual environment, because you want to test the *installed* version of your package, not the local code from the repository.


Although I found that there are some packages developing tests with `__init__.py` inside a `tests` directory, they are using *absolute imports* instead. (e.g. [coverage.py][case_covrage.py], [requests][case_requests])

I haven't dug deeper into those cases, but my first guess is that we are just running different development workflow, and it isn't harmful.


Besides, when it comes to the discussion about using `src` or `non-src` layout for packaging, I have no preference. Just like the same author of the article referred in that pytest documentation:

> Any layout will work. Mistakes can be corrected. - [Rehashing the src layout](https://blog.ionelmc.ro/2017/09/25/rehashing-the-src-layout/)


In conclusion, I like this revision. 👍 Though the import statement might be a little bit of ambiguous, it works well in current workflow (with `pip install -e .[tests]` things). And I'll keep this in mind in case there is a better design to remove this little flaw in the future.


[case_covrage.py]: https://github.com/nedbat/coveragepy/blob/7a4292dcd68e7b9d4d426d9f2eaf6f0535cdd97f/tests/test_api.py#L24
[case_requests]: https://github.com/psf/requests/blob/bfb93d4b7d269a8735f1b216093e7e9a9fdc4517/tests/test_testserver.py#L9
